### PR TITLE
internal/integration: speed up test execution

### PR DESCRIPTION
**Pull request by masseelch**
Integration tests are now run each in its own job.

I had to do some kind of "hacky" solution, so it working correctly in the CI by using an env var when running the test. This does not affect local running.

### CODE REVIEW

File: internal/ci/main.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"log"
+	"os"
+	"text/template"
+)
+
+//go:generate go run main.go
+
+// Job defines an integration job to run.
+type Job struct {
+	Version string   // version to test (used as GO_TEST_ONLY_VERSION env)
```
Comment (a8m) [inline internal/ci/main.go:15]: Do you still need this env?

File: internal/integration/integration_test.go [hunk 2]
```diff
@@ -21,6 +23,22 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestMain(m *testing.M) {
+	var service string
+	flag.StringVar(&service, "service", "", "[mysql56, postgres10, tidb5, ...] what version to test")
```
Comment (a8m) [inline internal/integration/integration_test.go:28]: `dialect` or `database` feel like a better name imo.

File: internal/integration/mysql_test.go [hunk 3]
```diff
@@ -28,22 +29,21 @@ type myTest struct {
...
 
-var myTests struct {
-	sync.Once
-	drivers map[string]*myTest
-}
+var (
+	myTests struct {
+		drivers map[string]*myTest
+	}
+	myPorts = map[string]int{
```
Comment (a8m) [inline internal/integration/mysql_test.go:36]: you can keep it under the `myTests` variable.

File: internal/integration/script_test.go [hunk 4]
```diff
@@ -174,11 +174,15 @@ func (t *myTest) cmdCmpShow(ts *testscript.TestScript, _ bool, args []string) {
 
 func (t *pgTest) cmdCmpShow(ts *testscript.TestScript, _ bool, args []string) {
 	cmdCmpShow(ts, args, func(schema, name string) (string, error) {
-		buf, err := exec.Command("docker", "ps", "-qa", "-f", fmt.Sprintf("name=postgres%s", t.version)).CombinedOutput()
+		buf, err := exec.Command("docker", "ps", "-qa", "-f", fmt.Sprintf("publish=%d", t.port)).CombinedOutput()
```
Comment (a8m) [inline internal/integration/script_test.go:177]: Can you elaborate on why filter by port and not by name?

**Review by a8m — APPROVED** [review]
Looks good, but please see questions before landing.

File: internal/integration/script_test.go (hunk 4 above)
Comment (a8m) [reply internal/integration/script_test.go:177]: Got the answer on our chat.

File: internal/ci/main.go (hunk 1 above)
Comment (masseelch) [reply internal/ci/main.go:15]: No. I'll change the docs

**Pull request merged**